Sources/SwiftDocC/DocumentationService/Convert/Fallback Link Resolution/ConvertServiceFallbackResolver.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A resolver that attempts to resolve local references that wasn't included in the catalog or symbol input.
+///
+/// The ``ConvertService`` builds documentation for a single page at a time. If this page's content contains references to other local symbols, pages, or
+/// assets that aren't included in the original ``ConvertRequest``, this fallback resolver resolves those references.
```

Comment:
How is possible (in which scenario) can exist symbols/assets that were not part of the original convert request? how are these injected?

Reply:
Like this documentation says: 
> The ``ConvertService`` builds documentation for a single page at a time.

For example, if you have `SomeModule` that contains a `SomeClass` which contains `someFunction()` and the ConvertService builds documentation for `someFunction()`, the ConvertService will pass a "partial symbol graph file" that contains `someFunction()` but doesn't contain `SomeClass` or any other symbols in the module. 

>If this page's content contains references to other local symbols, pages, or assets that aren't included in the original ``ConvertRequest``, this fallback resolver resolves those references.

The symbol graph file will still have relationships that say that `someFunction()` "is a member of" `SomeClass`. When DocC encounters the relationship with the USR for `SomeClass` but can't find it in the symbol graph file, it will pass the USR to the convert service and ask it to resolve the symbol. The convert service will then on-demand fill in the referenced symbol so that DocC can render the page for `someFunction()` and display "SomeClass" where it is referenced.

Similarly, if `someFunction()` curates another local page that wasn't included in the original request, DocC will pass the link to the convert service so that it can on-demand fill in the missing content. In both these cases DocC will only render a page for `someFunction()` so both the resolved symbol and the resolved link are "external" to this documentation build.

An external asset could happen if the curated external symbol had a `@PageImage` that should be displayed inline on the `someFunction()` page. Like before, DocC will ask the convert service to on-demand fill in the asset that wasn't part of the original request.

Reply:
I added a version of this information in https://github.com/apple/swift-docc/pull/806/commits/aba61b3dc8a2e0a5fcce03583c64c5bd42a318bd